### ISSUE TITLE
weights pruning with tfmot

### ISSUE THREAD

**Issue by Kaede93** [tensorflow/model-optimization#570]
I'm trying the [official demo](https://tensorflow.google.cn/model_optimization/guide/pruning/pruning_with_keras) of tfmot for keras model weights pruning, I followed every steps of the guideline but there are some problems confusing me.

1. Why the size of pruned model (.h5) doesn't change unless compressing to .gz? Is that means the weights pruning can not accelerating the inference (almost same inference time using model and model_for_export to do inference with 10K images from MNIST test set) but only reducing the storage space?

2. I used model.get_weights() for getting weight matrix for each layer and found that the weight matrix of model and model_for_export were identical. I think this is abnormal because the size of model (in .gz format) was reduced, the weights of model_for_export should be more sparse. Can you give me some help?

thank you for your time

**Comment by alanchiao** [comment]
Did you see the comments at https://tensorflow.google.cn/model_optimization/guide/pruning/pruning_with_keras#create_3x_smaller_models_from_pruning?

It tries to explain the things that confuse you.  Specifically: "Applying a standard compression algorithm is necessary since the serialized weight matrices are the same size as they were before pruning. However, pruning makes most of the weights zeros, which is added redundancy that algorithms can utilize to further compress the model."

https://tensorflow.google.cn/model_optimization/guide/pruning#api_compatibility_matrix suggests that latency improvements are on the way. You can imagine that so long as there a 0s in the matrices, any tool could recognize that and create a sparse form of it to improve inference.

**Comment by teijeong** [comment]
Closing as the question seems to be answered.

**Comment by STAROFWIND** [comment]
Hi all, any updates for these questions ? 
I have the same confusion that after pruning, I have the same size model and the same inference time as before pruning. 
So it doesn't make sense right ? I mean pruning model is not useful ( have no advantage) or I miss something here ? 
Can you tell me what can I do next with a pruned model ? like how to cut (remove) the zero-weights of the pruned model to reduce size and inference time ?
Thank you so much.
